### PR TITLE
Added ompl_trajectory converter

### PR DESCRIPTION
**Pull request by ChamzasKonstantinos**
Converts an omplGeometricPath to a moveitTrajectory

### CODE REVIEW

File: robowflex_ompl/include/robowflex_ompl/ompl_trajectory.h [hunk 1]
```diff
@@ -53,6 +53,13 @@ namespace robowflex
              *  \return The Geometric Path equivalent of the trajectory.
              */
             ompl::geometric::PathGeometric toOMPLPath(const ompl::geometric::SimpleSetupPtr &ss);
+
+            /** \brief Sets the trajectory from an OMPL Path and a reference state.
+             *  \param[in] reference_state A full state that contains the values for all the joints.
+             *  \param[in] path The geometric ompl path to conver.
```
Comment (zkingston) [inline robowflex_ompl/include/robowflex_ompl/ompl_trajectory.h:59]: conver -> convert

File: robowflex_ompl/include/robowflex_ompl/ompl_trajectory.h [hunk 2]
```diff
@@ -53,6 +53,13 @@ namespace robowflex
              *  \return The Geometric Path equivalent of the trajectory.
              */
             ompl::geometric::PathGeometric toOMPLPath(const ompl::geometric::SimpleSetupPtr &ss);
+
+            /** \brief Sets the trajectory from an OMPL Path and a reference state.
+             *  \param[in] reference_state A full state that contains the values for all the joints.
```
Comment (zkingston) [inline robowflex_ompl/include/robowflex_ompl/ompl_trajectory.h:58]: Would it not be reasonable to have the OMPL trajectory class itself contain the reference state for the trajectory? This trajectory was generated with respect to some default state. This reference state could be the zero state by default. Is there a case where we want an OMPL trajectory and do not have a reference state (even implicitly)?

File: robowflex_ompl/src/ompl_trajectory.cpp [hunk 3]
```diff
@@ -33,3 +33,17 @@ ompl::geometric::PathGeometric OMPL::OMPLTrajectory::toOMPLPath(const ompl::geom
 
     return path;
 }
+
+void OMPL::OMPLTrajectory::fromOMPLPath(const robot_state::RobotState &reference_state,
+                                        const ompl::geometric::PathGeometric &path)
+{
+    const auto &mbss = std::dynamic_pointer_cast<ompl_interface::ModelBasedStateSpace>(
```
Comment (zkingston) [inline robowflex_ompl/src/ompl_trajectory.cpp:40]: Might want to throw in an error check to see if mbss is null in case a junk path is thrown in.

**Review by zkingston — CHANGES_REQUESTED** [review]
LGTM, just a few nits / questions.

File: robowflex_ompl/include/robowflex_ompl/ompl_trajectory.h (hunk 2 above)
Comment (ChamzasKonstantinos) [reply robowflex_ompl/include/robowflex_ompl/ompl_trajectory.h:58]: This is a big change making it an issue,

File: robowflex_ompl/src/ompl_trajectory.cpp [hunk 4]
```diff
@@ -33,3 +33,23 @@ ompl::geometric::PathGeometric OMPL::OMPLTrajectory::toOMPLPath(const ompl::geom
...
+        path.getSpaceInformation()->getStateSpace());
+
+    if (not mbss)
+    {
+        ROS_ERROR("Failed to extract StateSpace from provided OMPL path!");
+        return;
```
[the remark this answers is not in the document]
Comment (ChamzasKonstantinos) [reply robowflex_ompl/src/ompl_trajectory.cpp:46]: You mean an exception? because I am already throwing an error

**Review by zkingston — APPROVED** [review]

**Pull request merged**